### PR TITLE
feat: Check profitability before requesting batches (mvp)

### PR DESCRIPTION
**Pull request by facundomedica**
Add a profitability check for the batch requester loop.
- This will help us create bigger batches from the get-go, instead of having to wait to have a big amount of txs piling up.
- Also it will allow us to give more accurate estimations on the FE while bridging

MVP because we will be adding a more complex solution sometime down the road, in which we get the gas costs from past transactions.

### CODE REVIEW

File: orchestrator/main_loops.go [hunk 1]
```diff
@@ -275,17 +278,38 @@ func (p *gravityOrchestrator) EthSignerMainLoop(ctx context.Context) (err error)
 func (p *gravityOrchestrator) BatchRequesterLoop(ctx context.Context) (err error) {
 	logger := p.logger.With().Str("loop", "BatchRequesterLoop").Logger()
 
+	var estimatedGasCosts = []int64{
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:281]: 1. Can we move this outside the function?
2. Can we also add a small godoc explaining what this is, how it's used, etc...?

File: orchestrator/main_loops.go [hunk 2]
```diff
@@ -294,6 +318,39 @@ func (p *gravityOrchestrator) BatchRequesterLoop(ctx context.Context) (err error
 
 				unbatchedTokensWithFees = batchFeesResp.GetBatchFees()
 
+				if p.relayer.GetProfitMultiplier() != 0.0 {
+					gasPrice, err = p.ethProvider.SuggestGasPrice(context.Background())
+					if err != nil {
+						return fmt.Errorf("failed to get Ethereum gas estimate: %w", err)
+					}
+
+					usdEthPrice, err := p.priceFeeder.QueryUSDPriceByCoinID("ethereum")
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:327]: put `"ethereum"` in a const?

File: orchestrator/main_loops.go [hunk 3]
```diff
@@ -275,17 +278,38 @@ func (p *gravityOrchestrator) EthSignerMainLoop(ctx context.Context) (err error)
...
+		usdEthPriceDec := decimal.NewFromFloat(0.0)
+		gasPrice := big.NewInt(0)
+		tokensPrices := make(map[string]decimal.Decimal)
+		tokensDecimals := make(map[string]uint8)
+
 		pg.Go(func() error {
 			var unbatchedTokensWithFees []types.BatchFees
 
 			if err := retry.Do(func() (err error) {
+
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:312]:
```suggestion
```

File: orchestrator/main_loops.go [hunk 4]
```diff
@@ -314,10 +371,35 @@ func (p *gravityOrchestrator) BatchRequesterLoop(ctx context.Context) (err error
 					return nil
 				}
 
-				logger.Info().Str("token_contract", tokenAddr.String()).Str("denom", denom).Msg("sending batch request")
+				shouldRequestBatch := true
+
+				if p.relayer.GetProfitMultiplier() != 0.0 {
+					// First we get the cost of the transaction in USD
+					totalETHcost := big.NewInt(0).Mul(gasPrice, big.NewInt(estimatedGasCosts[unbatchedToken.TxCount]))
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:378]: Do we need to do a bounds check?

Comment (facundomedica) [reply orchestrator/main_loops.go:378]: We don't need to, as it's guaranteed by the chain that the max amount of txs will be 100. That being said, I just realize there's a bug here lol (should be `TxCount - 1`). Will fix

File: orchestrator/main_loops.go (hunk 1 above)
Comment (facundomedica) [reply orchestrator/main_loops.go]: Done ✅

File: orchestrator/main_loops.go (hunk 2 above)
Comment (facundomedica) [reply orchestrator/main_loops.go]: ✅

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**